### ISSUE TITLE
Zinc randomly fails when retrieving images

### ISSUE THREAD

**Issue by peteruhnak** [pharo-project/pharo-project-contributors#2]
Zinc randomly fails when retrieving images.

Maybe they shouldn't be regenerated on every run and should be cached.

And if the retrieval fails then use the default one instead of killing the whole build.

**Comment by svenvc** [comment]
as far as I remember, they images are cached in files next to the image, but that won't help if you do a clean build ;-)

**Comment by peteruhnak** [comment]
Yes, jenkins is doing clean build and I am not sure how I can keep files
between builds (or I can check against the production site, which would be
easier). In either case it shouldn't kill the whole build.

**Comment by svenvc** [comment]
You have to understand this started as a quick hack that I ran manually on my machine, once in a while. It was not written to be run unattended on a server.

The problem is that you cannot check the avatars each time, that is too much work. And checking a URL to know if something changes is just as much work. We need a robust cache, checking the URL only after X days.

The main loop also needs to be more robust, obviously.

> On 02 Mar 2016, at 17:11, Peter Uhnak notifications@github.com wrote:
> 
> Yes, jenkins is doing clean build and I am not sure how I can keep files
> between builds (or I can check against the production site, which would be
> easier). In either case it shouldn't kill the whole build.
> 
> On Wed, Mar 2, 2016 at 5:08 PM, Sven Van Caekenberghe <
> notifications@github.com> wrote:
> 
> > as far as I remember, they images are cached in files next to the image,
> > but that won't help if you do a clean build ;-)
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/pharo-project/pharo-project-contributors/issues/2#issuecomment-191305326
> > .
> > 
> > —
> > Reply to this email directly or view it on GitHub.

## 

Sven Van Caekenberghe - mailto:sven@beta9.be
Beta Nine - software engineering - http://www.beta9.be

**Comment by peteruhnak** [comment]
What is the reasoning behind caching the images in the first place? Why not just hotlink them / let gravatar handle it?

Another option is to not worry about this at all and delegate the image handling some server-side script (so the request would be intercepted and cached on-demand).

**Comment by svenvc** [comment]
> On 02 Mar 2016, at 23:29, Peter Uhnak notifications@github.com wrote:
> 
> What is the reasoning behind caching the images in the first place? Why not just hotlink them / let gravatar handle it?

What is implemented is that people could give a URL to any image and that all of those images would be converted to the same size and format.

> Another option is to not worry about this at all and delegate the image handling some server-side script (so the request would be intercepted and cached on-demand).

That requires a app/server.

**Comment by peteruhnak** [comment]
> What is implemented is that people could give a URL to any image and that all of those images would be converted to the same size and format.

Same size can be easily achieved with css/html; also you can ask gravatar (that's also being used) for specific sizes directly.
I feel like it's a lot of work to cache it locally with no real benefits, unless I am missing something obvious.

> That requires a app/server.

files.pharo.org where it will be hosted supports at least PHP.

**Comment by PharoProject** [comment]
Just change it any way you want, I am happy that you took this on you, even simple things quickly become a lot of work, no need for this discussion, make it as easy for yourself as possible
